### PR TITLE
suggestions for cep2 refinement

### PR DESCRIPTION
**Pull request by gidden**
A concrete proposal w.r.t. https://groups.google.com/forum/#!topic/cyclus-dev/ml5zMNVc05Y

### CODE REVIEW

**Comment by rwcarlsen** [comment]
These changes suggest that the kernel should be responsible for providing some canonical level of domain general systems modeling capability.  I think that is out of scope for the core and believe that we should only provide the minimum necessary for testing.

I like the wording of the original better - it basically says that domain specific stuff doesn't go into the kernel (and kernel repo).  However, this doesn't mean that all domain-general stuff should go into the kernel.  And it also doesn't mean that domain general stuff can't go in cycamore.

Just my 2 satoshi (keeping all those negatives and positives straight is tricky business :-).

File: source/cep/cep2.rst [hunk 1]
```diff
@@ -96,18 +97,30 @@ of the following categories:
...
+for housing general archetypes that meet this need. Supply chain network models
+require, at minimum, three types of entities (i.e., nodes in a network)
+:cite:`bazaraa2011linear`:
+
+* Sources (Supply)
+* Transshipment (Storage)
+* Sinks (Demand)
+
+In addition to enabling such basic simulations, the dynamic resource exchange
+simulation concern is also responsible for all optimizations of the fuel cycle.
```
Comment (scopatz) [inline source/cep/cep2.rst]: This seems very strongly worded And unenforceable.

Comment (gidden) [inline source/cep/cep2.rst]: I don't disagree; however it was already there, author =). I'd be happy to
remove it.

On Thu, Apr 9, 2015 at 3:43 PM, Anthony Scopatz <notifications@github.com>
wrote:

> In source/cep/cep2.rst
> <https://github.com/cyclus/cyclus.github.com/pull/129#discussion_r28100585>
> :
>
> > +allowed to discriminate between any of these specializations nor perform
> > +domain-specific computations (such as transmuting a material or pricing a
> > +derivative).
> > +
> > +In broad terms, this concern *fully enables* the general, dynamic simulation of
> > +supply chain networks in a resource-neutral manner. As such it is responsible
> > +for housing general archetypes that meet this need. Supply chain network models
> > +require, at minimum, three types of entities (i.e., nodes in a network)
> > +:cite:`bazaraa2011linear`:
> > +
> > +* Sources (Supply)
> > +* Transshipment (Storage)
> > +* Sinks (Demand)
> > +
> > +In addition to enabling such basic simulations, the dynamic resource exchange
> > +simulation concern is also responsible for all optimizations of the fuel cycle.
>
> This seems very strongly worded And unenforceable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus.github.com/pull/129/files#r28100585>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

File: source/cep/cep2.rst [hunk 2]
```diff
@@ -3,8 +3,9 @@ CEP 2 - Separation of Concerns in the |cyclus| Ecosystem
 
 :CEP: 2
 :Title: Separation of Concerns in the |cyclus| Ecosystem, or Where Do I Put My Code?
-:Last-Modified: 2013-08-27
+:Last-Modified: 2015-04-09
 :Author: Anthony Scopatz
+:Edited By: Matthew Gidden
```
Comment (scopatz) [inline source/cep/cep2.rst]: "Edited By" is not a valid key. See CEP1.  Probably should just add your name to the Author tag

File: source/cep/cep2.rst (hunk 1 above)
Comment (scopatz) [inline source/cep/cep2.rst]: Probably should keep it then.

File: source/cep/cep2.rst [hunk 3]
```diff
@@ -96,18 +97,30 @@ of the following categories:
...
-between multiple actors as well as finding the initial conditions which optimize 
-a value function subject to some constraints.  
+specializations.  For example, materials and U.S. Dollars are both sub-types of
+resources. Still, the dynamic resource exchange simulation concern is not
+allowed to discriminate between any of these specializations nor perform
+domain-specific computations (such as transmuting a material or pricing a
+derivative).
+
+In broad terms, this concern *fully enables* the general, dynamic simulation of
+supply chain networks in a resource-neutral manner. As such it is responsible
```
Comment (scopatz) [inline source/cep/cep2.rst]: This first sentence is confusing and probably isn't needed.

File: source/cep/cep2.rst [hunk 4]
```diff
@@ -96,18 +97,30 @@ of the following categories:
...
+for housing general archetypes that meet this need. Supply chain network models
+require, at minimum, three types of entities (i.e., nodes in a network)
```
Comment (scopatz) [inline source/cep/cep2.rst]: `(i.e., nodes in a network)` is redundant

File: source/cep/cep2.rst [hunk 5]
```diff
@@ -96,18 +97,30 @@ of the following categories:
...
+:cite:`bazaraa2011linear`:
+
+* Sources (Supply)
+* Transshipment (Storage)
```
Comment (scopatz) [inline source/cep/cep2.rst]: I don't think we should enshrine the `Transshipment` name in a CEP.  It is fine to note that in the literature they are sometimes called Transhipment nodes, I would not want to see a PR that came through with an archetype whose name is this, and justification is this CEP.

**Comment by scopatz** [comment]
Hi @gidden - I have put in my comments in-line.  I think with these edits it could be accepted.  However, I think that modifications like this should go through the full CEP review process.  An email should go out to the mailing list after you have updated the text.

**Comment by gidden** [comment]
updated

**Comment by scopatz** [comment]
Thanks @gidden!

**Pull request merged**